src/rastervision/label_stores/segmentation_raster_file.py

```diff
@@ -116,59 +116,41 @@ def set_labels(
             raise ValueError('Unsure how to handle source={}'.format(
                 type(source)))
 
-    def interesting_subwindow(self, window: Box, size: int,
-                              shift: int) -> Union[Box, None]:
-        """Given a larger window, return a sub-window that contains interesting
-        pixels (pixels of class 1).
+    def window_predicate(self, window: Box, target_classes: List[int]) -> bool:
+        """Given a window, answer whether the window contains enough pixels in
+        the target classes.
 
         Args:
              window: The larger window from-which the sub-window will
                   be clipped.
-             size: The linear size (height and width) of the smaller
-                  window.
-             shift: How far to shift the returned window to the
-                  right.  This is useful because the returned window
-                  is constructed by finding the first pixel from the
-                  top-left, which can result in objects being cutoff.
+             target_classes: The classes of interest.  The given
+                  window is examined to make sure that it contains a
+                  sufficinet number of interesting pixels.
 
         Returns:
-             Either a sub-window containing interesting pixels or None
-             if no such window can be found.
+             True (the window does contain interesting pixels) or False.
 
         """
         if self.source is not None:
-            larger_size = window.xmax - window.xmin  # XXX assumed square
+            size = window.xmax - window.xmin  # Assumed square
 
             labels = self.source._get_chip(window)
             r = np.array(labels[:, :, 0], dtype=np.uint32) * (1 << 16)
             g = np.array(labels[:, :, 1], dtype=np.uint32) * (1 << 8)
             b = np.array(labels[:, :, 2], dtype=np.uint32) * (1 << 0)
             packed = r + g + b
             translated = self.source_to_rv(packed)
-            argmax = np.argmax(translated == 1)
-
-            if translated.sum() < (larger_size * math.sqrt(larger_size)):
-                return -2
-            elif argmax == 0 and not translated[0, 0] == 1:
-                return -1
-            elif size == larger_size:
-                return window
+
+            target_count = 0
+            for i in target_classes:
+                target_count = target_count + (translated == i).sum()
+
+            if target_count < (size * math.sqrt(size)):
```

Comment:
I'm not sure that `size * sqrt(size)` counts as "enough". If the only target class is cars for example, the `target_count` could be less than this. Maybe there should be an option for this, and it should be a threshold on the proportion of target pixels, similar to `ioa_thresh`?